info-compiler/src/test/java/com/google/sps/InfoCompilerTest.java

```diff
@@ -453,6 +475,69 @@ public void compileInfo_discardTestElection2000()
     assertThat(queryResult.hasNext()).isFalse();
   }
 
+  @Test
+  public void compileInfo_infoExpires()
+      throws IOException {
+    // Execute the entire information compilation process and set {@code DATA_EXPIRATION_SECONDS}
+    // such that all newly added would be considered outdated immediately and thus cleared.
+    Timestamp past = Timestamp.now();
+    JsonObject electionJsonCopy = electionJson.deepCopy();
+    JsonObject election =
+        ((JsonObject) electionJsonCopy.getAsJsonArray("elections").get(0));
+    election.addProperty("ocdDivisionId", "ocd-division/country:us/state:" + STATE.toLowerCase());
+    InfoCompiler infoCompilerMock = mock(InfoCompiler.class);
+    // Any new data will be seen as outdated and thus cleared.
+    infoCompilerMock.DATA_EXPIRATION_SECONDS = -1;
+    infoCompilerMock.datastore = this.datastore;
+    infoCompilerMock.addresses = Arrays.asList(ADDRESS);
+    infoCompilerMock.webCrawler = mock(WebCrawler.class);
+    doCallRealMethod().when(infoCompilerMock).compileInfo();
+    doCallRealMethod().when(infoCompilerMock).queryAndStoreBaseElectionInfo();
+    doCallRealMethod().when(infoCompilerMock).queryAndStoreElectionContestInfo();
+    doCallRealMethod()
+        .when(infoCompilerMock).queryAndStoreElectionContestInfo(anyString(), anyString());
+    doCallRealMethod().when(infoCompilerMock).queryAndStore(anyString(), anyString(), anyObject());
+    doCallRealMethod().when(infoCompilerMock).storeBaseElectionInDatabase(anyObject());
+    doCallRealMethod()
+        .when(infoCompilerMock).storeElectionContestInDatabase(anyString(), anyObject());
+    doCallRealMethod()
+        .when(infoCompilerMock)
+            .storeElectionContestCandidateInDatabase(anyObject(), anyObject(), anyObject());
```

Comment:
Instead of  specifying every method to use the real method, you can use a mock spy and only override the behavior of `queryCivicInformation()`:
https://www.vogella.com/tutorials/Mockito/article.html#mockito_spy

Reply:
Gotcha thank you! I'll use a mock spy.